### PR TITLE
Remove dependency on sidekiq

### PR DESCRIPTION
**Pull request by phawk**
Remove dependency on sidekiq and use sucker_punch for background processing

Why:

Using sidekiq requires us to run a worker dyno on heroku

This change addresses the need by:

Allowing us to simplify hosting and running costs at the cost of more fragile background processing

### CODE REVIEW

File: lib/attachment_processors/image.rb [hunk 1]
```diff
@@ -24,7 +25,7 @@ def resize_and_upload(width: 600, height: 400, tag: :thumb)
       tmp_img = Tempfile.new([SecureRandom.hex(18), ext])
 
       # resize it
-      img = Magick::Image.read(attachment.url).first
+      img = Magick::Image.from_blob(open(attachment.url).read).first
```
Comment (phawk) [inline lib/attachment_processors/image.rb]: For some reason rmagick wasn’t reading the image from URL locally, needed to read the raw data. Seen a few articles where certain versions of imagemagick dont support opening a file by url.

File: lib/tasks/attachments.rake [hunk 2]
```diff
@@ -0,0 +1,14 @@
+namespace :attachments do
```
Comment (phawk) [inline lib/tasks/attachments.rake]: Allows us to catch processing attemps that have failed as sucker punch won’t retry an item. We can set this to run hourly

**Comment by colinyoung87** [comment]
How come we don't need the s3 direct upload any more?

**Comment by colinyoung87** [comment]
:+1:

**Pull request merged**